pkg/operator/ceph/cluster/osd/pod_test.go

```diff
@@ -194,6 +194,7 @@ func TestStorageSpecConfig(t *testing.T) {
 	job, err := c.makeJob(n.Name, n.Devices, n.Selection, c.Storage.Nodes[0].Resources, storeConfig, metadataDevice, n.Location)
 	assert.NotNil(t, job)
 	assert.Nil(t, err)
+	assert.Equal(t, "rook-ceph-osd-prepare-node1", job.ObjectMeta.Name)
```

Comment:
is it possible to add a test case that exercises the truncating?  these shorter names won't cause the condition to be hit.

Reply:
I added a test to the `k8sutil` package, see https://github.com/galexrt/rook/blob/fix_1907/pkg/operator/k8sutil/k8sutil_test.go